builder/test/pipeline.cc

```diff
@@ -662,14 +663,13 @@ TEST_P(multiple_outputs, pipeline) {
   // For a 3D input in(x, y, z), compute sum_x = sum(input(:, y, z)) and sum_xy = sum(input(:, :, z)) in one stage.
   func::callable<const int, int, int> sum_x_xy = [](const buffer<const int>& in, const buffer<int>& sum_x,
                                                      const buffer<int>& sum_xy) -> index_t {
-    assert(sum_x.dim(1).min() == sum_xy.dim(0).min());
-    for (index_t z = sum_xy.dim(0).min(); z <= sum_xy.dim(0).max(); ++z) {
-      sum_xy(z) = 0;
-      for (index_t y = sum_x.dim(0).min(); y <= sum_x.dim(0).max(); ++y) {
-        sum_x(y, z) = 0;
+    for (index_t z = in.dim(2).min(); z <= in.dim(2).max(); ++z) {
```

Comment:
We can't use input bounds in callbacks.

Basically what is happening here now is you iterate over the entire input every time, even if you only write a small portion of the output.

This needs to be:
```
for (index_t z = std::min(sum_xy.dim(0).min(), sum_x.dim(1).min()); z <= std::max(sum_xy.dim(0).max(), sum_x.dim(1).max()); ++z) {
```

Reply:
> We can't use input bounds in callbacks.

Did not realize that. Fixed.